### ISSUE TITLE
No Version control systems found error if gitpython not installed

### ISSUE THREAD

**Issue by livia-b** [open-research/sumatra#270]
The first time I tried 

    smt init myproject

in a git repository I got **No Version control systems found**, I had to manually install gitpython (I guess it's not in the package dependencies). I had installed sumatra  on RHEL (in a virtual python enviroment)  with 

    easy_install sumatra

**Comment by Felix11H** [comment]
Had the same issue. For now

    pip install gitpython

is an easy way to get the missing package.

**Comment by apdavison** [comment]
This is the intended behaviour. The idea is not to automatically install dependencies that a user may not need (e.g. if they are using Mercurial or Subversion).

However, I am open to changing this. For example we could install at least the Git and Mercurial bindings by default.

Alternatively, we could just have a better error message, e.g. with instructions for how to install the required dependencies.

Any comments?

**Comment by timtroendle** [comment]
Given the popularity of ``Git`` and the fact that one needs at least one version control system for sumatra to work, ``GitPython`` could be a regular dependency.

On the other hand Sumatra already has quite some dependencies and so having an unnecessary one is not that favorable.

In any case I think the documentation in ``README.rst`` could be improved, stating that one _has to_ install one version control dependency and also mentioning the handy setuptools extras mechanism. 

So instead of

     pip install sumatra

it could state

    pip install sumatra[git]
    pip install sumatra[hg]
    ...

**Comment by maxalbert** [comment]
I'd be strongly in favour of a better error message. This issue keeps tripping me up on fresh systems even though I know what's causing it. We should rephrase the error message stating the possible causes of the error and making it clear that the reason for not finding any version control systems may be that the Python bindings for that version control system couldn't be found, rather than no version control system being present on the computer (which is what the error message sounds like) or the Sumatra project dir not being version-controlled.

I tend to agree that installing Python bindings for the various version control systems by default may not be that favourable. On the other hand, they aren't really that big and if it saves a lot of users some hassle then it may be better to simply install them. I also like @timtroendle suggestion to improve instructions in the README file.

**Comment by livia-b** [comment]
I agree, if I hadn't found a discussion on google groups about this issue I would have never thought to install gitpython - the error message made me just think to check if I was in the correct folder.

**Comment by apdavison** [comment]
How about the following error message?
```
Error: No working copy found at /path/to/my/project
Tried: Mercurial, Bazaar
Not available: Subversion, Git
You may need to install Python bindings for your version control system
- see http://sumatra.readthedocs.org/en/latest/installation.html#installing-python-bindings-for-your-version-control-system
```

**Comment by livia-b** [comment]
It looks a good message to me

**Comment by timtroendle** [comment]
Looks good to me too. One minor comment: potentially rather say ``No bindings available: Subversion, Git`` to make that point a little bit clearer.

**Comment by apdavison** [comment]
> potentially rather say No bindings available: Subversion, Git to make that point a little bit clearer.

We can't generally distinguish at the moment between "bindings not available" and "underlying VCS not available".

**Comment by maxalbert** [comment]
+1 on the great new error message and on @timtroendle's clarification (I would even say "No Python bindings available" to make it even clearer what "bindings" means). @apdavison: Why can't we say that e.g. the git Python bindings are not available if importing the `git` module results in an `ImportError`?

**Comment by apdavison** [comment]
Since it is possible to install gitpython without having git installed, I had thought it was better not to be too specific about the source of the error in case we lead users down the wrong path. However, (i) it turns out that "smt init" works fine if gitpython is there but git is not and (ii) not having git installed is unlikely if someone is already using it for their project, so I guess I agree with @timtroendle's clarification after all.

**Comment by apdavison** [comment]
OK, I've implemented the new error message, and made a new release (0.7.2) which also includes fixes for #268 and #269

**Comment by daphnecor** [comment]
I just had the same problem, seems like the new error message is not implemented in version 0.7.4

```
Error: No version control systems found. Please see the documentation for information on installing the required packages.
```
